tests/integration/sync_test.go

```diff
@@ -8127,6 +8127,80 @@ func Test_Sync_Partials_Plugins(t *testing.T) {
 		require.Error(t, err)
 		assert.ErrorContains(t, err, "partial non-existent-partial for plugin rate-limiting-advanced: entity not found")
 	})
+
+	t.Run("partial linking fails if partial information is not provided properly", func(t *testing.T) {
+		mustResetKongState(ctx, t, client, dumpConfig)
+		err := sync("testdata/sync/038-partials/plugin-partial-no-ids.yaml")
+		require.Error(t, err)
+		assert.ErrorContains(t, err, "partial for plugin rate-limiting-advanced: partial ID or name is required")
+
+		err = sync("testdata/sync/038-partials/ill-formatted-partial.yaml")
+		require.Error(t, err)
+		assert.ErrorContains(t, err, "partial for plugin rate-limiting-advanced: missing required fields - name or id")
+	})
+
+	t.Run("partial linking works fine in case of nested plugin", func(t *testing.T) {
+		mustResetKongState(ctx, t, client, dumpConfig)
+		currentState, err := fetchCurrentState(ctx, client, dumpConfig)
+		require.NoError(t, err)
+
+		targetState := stateFromFile(ctx, t, "testdata/sync/038-partials/nested-plugin-partials.yaml", client, dumpConfig)
+		syncer, err := deckDiff.NewSyncer(deckDiff.SyncerOpts{
+			CurrentState: currentState,
+			TargetState:  targetState,
+
+			KongClient: client,
+		})
+		require.NoError(t, err)
+
+		stats, errs, changes := syncer.Solve(ctx, 1, false, true)
+		require.Empty(t, errs, "Should have no errors in syncing")
+		logEntityChanges(t, stats, changes)
```

Comment:
Are we making use of the stats anywhere? 👀

Reply:
We need to log the changes so that test debugging is easier if something blows up.